modules/node_groups/templates/userdata.sh.tpl

```diff
@@ -1,9 +1,20 @@
 #!/bin/bash -e
+%{ if length(ami_id) == 0 || ami_is_eks_optimized ~}
 
-# Allow user supplied pre userdata code
+sed -i '/^KUBELET_EXTRA_ARGS=/a KUBELET_EXTRA_ARGS+=" ${kubelet_extra_args}"' /etc/eks/bootstrap.sh
+%{endif ~}
+%{if length(pre_userdata) > 0 ~}
+
+# User supplied pre userdata code
 ${pre_userdata}
+%{endif ~}
+%{ if length(ami_id) > 0 && ami_is_eks_optimized ~}
```

Comment:
If think this should be 
```suggestion
%{ if length(ami_id) > 0 && ami_is_eks_optimized == false ~}
```

Shouldn't it?

Reply:
No, what's in the PR is correct. We want to call the bootstrap script if the `ami_id` is set and the AMI is an EKS optimised image. If the `ami_id` is set in a launch template then the launch template is used directly and not merged to create a managed launch template; in this case we can call `/etc/eks/bootstrtap.sh` ourselves if the image is an EKS optimised one with the script available.

Reply:
Oh, okay, so we need to add this even if we use an AWS supplied AMI here. I did not realised that.

Reply:
If you set `ami_id` you need this.